examples/provider/provider-config-with-secret-upbound-identity.yaml

```diff
@@ -0,0 +1,18 @@
+apiVersion: kubernetes.crossplane.io/v1alpha1
+kind: ProviderConfig
+metadata:
+  name: kubernetes-provider
+spec:
+  credentials:
+    source: Secret
+    secretRef:
+      namespace: crossplane-system
+      name: cluster-config
+      key: kubeconfig
+  identity:
+    type: UpboundToken
```

Comment:
@luxas any comment for this `identity.type`? Other supported values today are, `GoogleApplicationCredentials`, `AzureServicePrincipalCredentials` and `AzureWorkloadIdentityCredentials`. 

May be `UpboundTokens` (to be consistent with other with appended `s`)
Or `UpboundStaticTokens` if we expect to see tokens evolving in time as you pointed out in another comment.

Reply:
I didn't make any distinction between user tokens or robot tokens since the logic doesn't change.

Reply:
Renamed as `UpboundToken` -> `UpboundTokens` for now.